### PR TITLE
Issue #2701

### PR DESCRIPTION
**Pull request by meuhland**
Here's more or less the solution I've deployed on my end for that... I see that there is now 
header('Accept-Ranges: bytes') but I'm not seeing any code that reads from that specific offset in the file.

### CODE REVIEW

**Comment by AngelFQC** [comment]
@meuhland Please format your code according with the Coding Conventions https://github.com/chamilo/chamilo-lms/wiki/Coding-conventions#coding-conventions

Apparently the PR works. I tested with a 1.2Gb video and the video player in browser is seekable

File: main/inc/lib/document.lib.php [hunk 1]
```diff
@@ -276,7 +276,79 @@ public static function file_get_mime_type($filename)
...
+     * @param string $contentType
+     *
+     * @return false if file doesn't exist, true if stream succeeded
+     */
+	public static function smartReadFile($full_file_name, $filename, $contentType = 'application/octet-stream')
+	{
+	if (!file_exists($full_file_name))
+	{
+		header ("HTTP/1.1 404 Not Found");
+		return;
```
Comment (AngelFQC) [inline main/inc/lib/document.lib.php]: This should be`return false;`

File: main/inc/lib/document.lib.php [hunk 2]
```diff
@@ -276,7 +276,79 @@ public static function file_get_mime_type($filename)
...
+	}
+	
+	$size	= filesize($full_file_name);
+	$time	= date('r', filemtime($full_file_name));
+	
+	$fm		= @fopen($full_file_name, 'rb');
+	if (!$fm)
+	{
+		header ("HTTP/1.1 505 Internal server error");
+		return;
```
Comment (AngelFQC) [inline main/inc/lib/document.lib.php]: This should be `return false;`

File: main/inc/lib/document.lib.php [hunk 3]
```diff
@@ -373,6 +445,16 @@ public static function file_send_for_download(
...
+				case 'video/mp4':
+				case 'audio/mpeg':
+				case 'audio/mp4':
+				case 'audio/ogg':
+				case 'audio/webm':
+				case 'audio/wav':
+				case 'video/ogg':
+				case 'video/webm':		
+					self::smartReadFile($full_file_name,$filename,$contentType);
+					break;
```
Comment (AngelFQC) [inline main/inc/lib/document.lib.php]: Maybe `break` should be `exit;` to skeep the code below

Comment (meuhland) [reply main/inc/lib/document.lib.php]: Thanks I'll review all those early next week!

**Comment by ywarnier** [comment]
I'm taking over this PR and adding the missing changes.

**Comment by meuhland** [comment]
Thanks @ywarnier wrote that quickly to answer a request we had internally for our portal and was more or less just tossing that as a proof of concept... so it was crudely put together (at best).

**Comment by ywarnier** [comment]
Also sent to the master branch.

@meuhland No worries. Hopefully those PR will get incrementally better over time.

**Comment by meuhland** [comment]
@ywarnier Yeah mixing a bunch of programming languages and standards will do that to you but yeah I got the gist of them still need to brush up on the php PSR-2 ones (*hint* I'm not a php guy!)

**Comment by ywarnier** [comment]
Using a smart IDE will usually allow you to decide - by project - which standard to apply, and PSR is usually one of them. That helps a bit when moving from one language or standard to another.

**Comment by meuhland** [comment]
Guess I'll renew my license of PhpStorm license then...

**Pull request merged**